File: src/ObjectDumper/DebuggeeInteraction/ExpressionProviders/IExpressionProvider.cs
namespace ObjectDumper.DebuggeeInteraction.ExpressionProviders;

internal interface IExpressionProvider
{
    string GetTargetFrameworkExpressionText();
    string GetIsSerializerInjectedExpressionText();
    string GetSerializedValueExpressionText(string expression, string settings);
    string GetLoadAssemblyExpressionText(string serializerFileName);
}

File: src/object-dumper-vscode/src/debuggee_interaction/expressionProviders/iExpressionProvider.ts
export interface IExpressionProvider {
    getStringTypeAssemblyLocationExpressionText(): string;
    getIsSerializerInjectedExpressionText(): string;
    getSerializedValueExpressionText(expression: string, format: string, options: string): string;
    getLoadAssemblyExpressionText(serializerFileName: string): string;
  }